kafka-rest/src/main/java/io/confluent/kafkarest/controllers/TopicManager.java

```diff
@@ -63,28 +63,12 @@ CompletableFuture<Optional<Topic>> getTopic(
    */
   CompletableFuture<Optional<Topic>> getLocalTopic(String topicName);
 
-  /**
-   * Creates a new Kafka {@link Topic} with either partitions count and replication factor or
-   * explicitly specified partition-to-replicas assignments.
-   */
-  CompletableFuture<Void> createTopic(
```

Comment:
Is there a good reason to remove the overload? Below commit introduced the overload precisely to avoid breaking external clients.
https://github.com/confluentinc/kafka-rest/commit/3f1e7c1912d14f7175eda468b42b8072e0877a41#diff-15612e27820603f2efe1f27c21963e53492df05773f3c9b4489fc575582ced9c

Reply:
It is more important to note that I changed the return type of the method. This is required in order to pass out the replication factor and number of partitions returned by the AdminClient.

So I took the hit of refactoring all the tests (which the referenced PR was avoiding), and I have also checked as well as I can that there are no references to this method outside this repo. By removing the overload and just ending up with the newer of the method signatures, we will soon be sure whether there are references that I missed.

Reply:
@msn-tldr If you have an example of an external client using this interface, please tell me. I can see that the TopicManager methods for listing and getting topics are used outside, but I think that the createTopic is not.

Reply:
I don't, but its very hard to say for sure-ity no-one isn't using outside of Confluent.

Maybe mark the method as @Deprecated first, what do you think?

Reply:
Done